edb/edgeql/compiler/inference/cardinality.py

```diff
@@ -121,28 +127,36 @@ def _bounds_to_card(
     )
 
 
+def _card_unzip(
+    args: Iterable[qltypes.Cardinality],
+) -> tuple[tuple[CardinalityBound, ...], tuple[CardinalityBound, ...]]:
+    card = list(zip(*(_card_to_bounds(a) for a in args)))
```

Comment:
This line is very difficult to understand, what does it do?

Reply:
This is (believe it or not) a commonish idiom in python for "unzipping" a list of tuples (or of lists, or whatever).
`zip` takes an arbitrary number of "iterable" arguments, and so passing all of the tuples in a list-of-tuples to `zip` as arguments means that we zip all of the tuples together, which has the effect of unzipping the original list-of-tuples.

Downsides are that it is weird and confusing and that it doesn't really behave sensibly on an empty list. (Which it *can't* without knowing the arity of the tuples statically)